addons/html_editor/static/tests/sanitize.test.js

```diff
@@ -35,3 +40,30 @@ test("sanitize plugin should handle aria-label attribute with data-oe-aria-label
         contentAfter: `<p data-oe-aria-label="status">a[]</p>`,
     });
 });
+
+test("fixInvalidHTML should close self-closing elements", () => {
+    expectMarkup(fixInvalidHTML(markup`<t/>`)).toBe("<t></t>");
+    expectMarkup(fixInvalidHTML(markup`<t class="test"/>`)).toBe('<t class="test"></t>');
+    expectMarkup(fixInvalidHTML(markup`<a/>`)).toBe("<a></a>");
+    expectMarkup(fixInvalidHTML(markup`<a href="#"/>`)).toBe('<a href="#"></a>');
+    expectMarkup(fixInvalidHTML(markup`<strong/>`)).toBe("<strong></strong>");
+    expectMarkup(fixInvalidHTML(markup`<strong class="bold"/>`)).toBe(
+        '<strong class="bold"></strong>'
+    );
+    expectMarkup(fixInvalidHTML(markup`<span/>`)).toBe("<span></span>");
+    expectMarkup(fixInvalidHTML(markup`<span id="test"/>`)).toBe('<span id="test"></span>');
+    expectMarkup(
+        fixInvalidHTML(markup`<t t-out="object.name"/>asdf<t t-out="object.parner_id.name"/>`)
+    ).toBe('<t t-out="object.name"></t>asdf<t t-out="object.parner_id.name"></t>');
+});
```

Comment:
It would be nice to have one or two more specific cases:
- one with nested tags (with the same tag) to make sure it closes the correct one
- one with escaped content in attribute( and also one with escaped text content) to make sure it doesn't get "unescaped"

Reply:
Apparently, the `expectMarkup` is doing something weird on the input. It self-closes elements without children.

<img width="295" alt="Screenshot 2025-07-08 at 16 43 54" src="https://github.com/user-attachments/assets/f43c338d-2554-447e-9b07-53ec1eeb8f9d" />

In the above screenshot, you see the values for `expected` and `value`. They're different. However, they pass the `expectMarkup` test because it uses `formatXML` on the compared values and it returns the same value for both.

I had to change the test cases to this format:

```js
    expect(fixInvalidHTML(markup`<t/>`).toString()).toBe("<t></t>");
```